workflow/rocoto/workflow_tasks.py

```diff
@@ -1102,57 +1086,52 @@ def eupd(self):
 
         return task
 
-    def atmensanalprep(self):
-
-        dump_suffix = self._base["DUMP_SUFFIX"]
+    def atmensanlinit(self):
         gfs_cyc = self._base["gfs_cyc"]
-        dmpdir = self._base["DMPDIR"]
         do_gfs_enkf = True if self.app_config.do_hybvar and 'gfs' in self.app_config.eupd_cdumps else False
+        cycledef = 'gdas'
+        if self.cdump in ['gfs'] and do_gfs_enkf:
+            cycledef = 'gfs'
+            if gfs_cyc != 4:
+                cycledef = 'gdas'
```

Comment:
Not sure what you are trying to do here. If this needs to run every cycle even if the free forecast is not, you can just run on the `gdas` `cycledef` regardless of `cdump`.

Reply:
This section replicates, in part, what was in the removed `atmensanalprep` section.
```
        cycledef = self.cdump
        if self.cdump in ['gfs'] and do_gfs_enkf and gfs_cyc != 4:
            cycledef = 'gdas'
```
I assumed this was something we need to retain due to the option to now run a partial enkf cycle in the gfs as well as the full enkf cycle in the gdas.  I was not part of this change and so I don't know the details.  

The new `atmensanlinit` step needs to run after the prep step.  Its workflow setup is similar to the existing `eobs`.   I updated this section of `workflow_tasks.py` to minic `eobs`.  Done at 3a3d6288.

Reply:
Yeah, what was originally there was unnecessary too. If it needs to run every cycle regardless of how often the full forecast runs, the `cycledef` should be `gdas`, regardless of the job's actual `cdump` (or `gdas_half,gdas` if it also needs to run in the half-cycle).